test/index.test.js

```diff
@@ -1180,4 +1180,52 @@ describe('mongoose module:', function() {
       }
     });
   });
+
+  describe('createInitialConnection (gh-8302)', function() {
+    let m;
+
+    beforeEach(function() {
+      m = new mongoose.Mongoose();
+    });
+
+    afterEach(async function() {
+      await m.disconnect();
+    });
+
+    it('should delete existing connection when setting createInitialConnection to false (part 1)', function() {
```

Comment:
```suggestion
    it('should delete existing connection when setting createInitialConnection to false', function() {
```

as there is seemingly no `Part 2`